src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -45,10 +49,54 @@ cdef class RPCState:
             grpc_call_unref(self.call)
 
 
+# TODO(lidiz) inherit this from Python level `AioRpcStatus`, we need to improve
+# current code structure to make it happen.
+class AbortError(Exception): pass
+
+
+def _raise_if_aborted(RPCState rpc_state):
+    """Raise AbortError if RPC is aborted.
+
+    Server method handlers may suppress the abort exception. We need to halt
+    the RPC execution in that case. This function needs to be called after
+    running application code.
+    """
+    if rpc_state.abort_exception is not None:
+        raise rpc_state.abort_exception
+
+
+async def _perform_abort(RPCState rpc_state,
+                         grpc_status_code code,
+                         str details, 
+                         tuple trailing_metadata,
+                         object loop):
+    """Perform the abort logic.
+
+    Sends final status to the client, and then set the RPC into corresponding
+    state.
+    """
+    if rpc_state.abort_exception is not None:
+        raise RuntimeError('Abort already called!')
+    else:
+        # Keeps track of the exception object. After abort happen, the RPC
+        # should stop execution. However, if users decided to suppress it, it
+        # could lead to undefined behavior.
+        rpc_state.abort_exception = AbortError('Locally aborted.')
```

Comment:
I'm wondering if `AbortError` could be always the same exception.... but not seeing much benefits on using a constant exception.

Reply:
The issue I'm trying to solve is that if users suppressed the exception, and continue whatever logic. We need a way to explicitly log an error saying, hey, you should not suppress the exception, and here is the location that you called abort (see `traceback`).